diff -rupN commons-net-2.0-original/src/main/java/org/apache/commons/net/ftp/FTPClient.java commons-net-2.0-modified/src/main/java/org/apache/commons/net/ftp/FTPClient.java
--- commons-net-2.0-original/src/main/java/org/apache/commons/net/ftp/FTPClient.java	2008-10-19 18:47:19.000000000 +0200
+++ commons-net-2.0-modified/src/main/java/org/apache/commons/net/ftp/FTPClient.java	2009-11-11 14:33:42.000000000 +0100
@@ -2260,8 +2260,13 @@ implements Configurable
                     // if a parserKey hasn't been supplied, and a configuration
                     // hasn't been supplied, then autodetect by calling
                     // the SYST command and use that to choose the parser.
-                    __entryParser = 
-                        __parserFactory.createFileEntryParser(getSystemName());
+
+                    // Maxence: getSystemName() returns null if the SYST command is not supported by the remote server.
+                    // In that case, we default to UNIX ; DefaultFTPFileEntryParserFactory will otherwise NPE.
+                    String systemName = getSystemName();
+                    __entryParser = __parserFactory.createFileEntryParser(
+                        systemName==null?FTPClientConfig.SYST_UNIX:systemName
+                    );
                 }
             }
         }
